### PR TITLE
Make data connection for S3 model sources AWS-unspecific

### LINKED ISSUE #1381: [Feature Request]: make data connection for S3 model sources AWS-unspecific
### Solution

Implementation details can be found in https://github.com/opendatahub-io/odh-dashboard/issues/1381#issuecomment-1607711576

---

### Feature description

Currently, at least the Workbench creation Data Connection GUI for model servers (to get model files from) is highly AWS S3 specific. I don't know whether that is intended, as models could also live in S3 compatible object storage on a company-internal server in a bucket, e.g. IBM Cloud Object Storage. 

Certificate trust is also an issue when, there are ways with trusted-ca-bundle configmap to get all trusted certs and use them as trusted when making https requests.

### Describe alternatives you've considered

_No response_

### Anything else?

Bildschirmfoto 2023-06-14 um 14 34 59

@Joohoo Unclear whether there is a connection with model-mesh as well, as dashboard is just the GUI side of things.

Pep Turró Mauri @codificat of Red Hat once created an issue for that, too, I think, not related to the GUI, but still

https://issues.redhat.com/browse/RHODS-8813

### PR DESCRIPTION
**Pull request by manaswinidas**
Fixes #1381 

## Description

Changes as suggested in [#1381 comment](https://github.com/opendatahub-io/odh-dashboard/issues/1381#issuecomment-1607711576)
Screenshot 2023-07-06 at 9 50 57 PM
Screenshot 2023-07-06 at 9 51 34 PM

## How Has This Been Tested?

1. Check out the PR branch
2. Try to create a data connection and/or pipeline server using the "Create New data connection" option. Also, look at the modal after clicking on "View Pipeline Configuration" while the pipeline is getting created and while Deploying a model using a "New Data Connection"

## Test Impact

Just `visual changes` as described in #1381 and added test for endpoint required field for "Deploy model"

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] The commits have meaningful messages (squashes happen on merge by the bot).
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/pipelines/content/ViewPipelineServerModal.tsx [hunk 1]
```diff
@@ -78,21 +78,19 @@ const ViewPipelineServerModal: React.FC<ViewPipelineServerModalProps> = ({
                   </DescriptionListDescription>
                 </DescriptionListGroup>
                 <DescriptionListGroup>
-                  <DescriptionListTerm>AWS_ACCESS_KEY_ID</DescriptionListTerm>
+                  <DescriptionListTerm>Access-Key</DescriptionListTerm>
                   <DescriptionListDescription>
-                    {objectStorageRecord?.AWS_ACCESS_KEY_ID}
+                    {objectStorageRecord?.['Access-Key']}
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/ViewPipelineServerModal.tsx:83]: dont access the keys like this `['Access-Key']`

use the enum: `AWS_KEYS.ACCESS_KEY_ID`

File: frontend/src/concepts/pipelines/content/ViewPipelineServerModal.tsx [hunk 2]
```diff
@@ -78,21 +78,19 @@ const ViewPipelineServerModal: React.FC<ViewPipelineServerModalProps> = ({
                   </DescriptionListDescription>
                 </DescriptionListGroup>
                 <DescriptionListGroup>
-                  <DescriptionListTerm>AWS_ACCESS_KEY_ID</DescriptionListTerm>
+                  <DescriptionListTerm>Access-Key</DescriptionListTerm>
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/ViewPipelineServerModal.tsx:81]:
```suggestion
                  <DescriptionListTerm>Access key</DescriptionListTerm>
```

File: frontend/src/concepts/pipelines/content/ViewPipelineServerModal.tsx [hunk 3]
```diff
@@ -78,21 +78,19 @@ const ViewPipelineServerModal: React.FC<ViewPipelineServerModalProps> = ({
...
                   <DescriptionListDescription>
-                    {objectStorageRecord?.AWS_ACCESS_KEY_ID}
+                    {objectStorageRecord?.['Access-Key']}
                   </DescriptionListDescription>
                 </DescriptionListGroup>
                 <DescriptionListGroup>
-                  <DescriptionListTerm>AWS_SECRET_ACCESS_KEY</DescriptionListTerm>
+                  <DescriptionListTerm>Secret-key</DescriptionListTerm>
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/ViewPipelineServerModal.tsx:87]:
```suggestion
                  <DescriptionListTerm>Secret key</DescriptionListTerm>
```

File: frontend/src/concepts/pipelines/content/ViewPipelineServerModal.tsx [hunk 4]
```diff
@@ -78,21 +78,19 @@ const ViewPipelineServerModal: React.FC<ViewPipelineServerModalProps> = ({
...
                   <DescriptionListDescription>
-                    <PasswordHiddenText
-                      password={objectStorageRecord?.AWS_SECRET_ACCESS_KEY ?? ''}
-                    />
+                    <PasswordHiddenText password={objectStorageRecord?.['Secret-Key'] ?? ''} />
```
Comment (Gkrumbach07) [inline frontend/src/concepts/pipelines/content/ViewPipelineServerModal.tsx:89]: same as above, use the enum

File: frontend/src/pages/projects/dataConnections/const.ts [hunk 5]
```diff
@@ -3,11 +3,11 @@ import { AWSDataEntry } from '~/pages/projects/types';
...
-  ACCESS_KEY_ID = 'AWS_ACCESS_KEY_ID',
-  SECRET_ACCESS_KEY = 'AWS_SECRET_ACCESS_KEY',
-  S3_ENDPOINT = 'AWS_S3_ENDPOINT',
-  DEFAULT_REGION = 'AWS_DEFAULT_REGION',
-  AWS_S3_BUCKET = 'AWS_S3_BUCKET',
+  ACCESS_KEY_ID = 'Access-Key',
+  SECRET_ACCESS_KEY = 'Secret-Key',
+  S3_ENDPOINT = 'Endpoint',
+  DEFAULT_REGION = 'Region',
+  AWS_S3_BUCKET = 'Bucket',
```
Comment (Gkrumbach07) [inline frontend/src/pages/projects/dataConnections/const.ts:10]: I am pretty sure these should not change. these keys need to have the AWS prefix still

> The underlying values that get injected into the Notebook and are passed to Model Serving will remain the same (prefixed with AWS). This is something we cannot change at this time. This is purely a visual set of changes.

File: frontend/src/pages/projects/dataConnections/const.ts [hunk 6]
```diff
@@ -60,10 +61,10 @@ export const EMPTY_AWS_SECRET_DATA: AWSDataEntry = [
   },
   {
     key: AWS_KEYS.S3_ENDPOINT,
-    value: 'https://s3.amazonaws.com/',
+    value: '',
   },
   {
     key: AWS_KEYS.DEFAULT_REGION,
-    value: 'us-east-1',
+    value: '',
```
Comment (Gkrumbach07) [inline frontend/src/pages/projects/dataConnections/const.ts:68]: value is not needed

File: frontend/src/pages/projects/dataConnections/const.ts [hunk 7]
```diff
@@ -60,10 +61,10 @@ export const EMPTY_AWS_SECRET_DATA: AWSDataEntry = [
   },
   {
     key: AWS_KEYS.S3_ENDPOINT,
-    value: 'https://s3.amazonaws.com/',
+    value: '',
```
Comment (Gkrumbach07) [inline frontend/src/pages/projects/dataConnections/const.ts:64]: value not needed

**Review by Gkrumbach07 — CHANGES_REQUESTED** [review]
i have not looked at tests yet.

File: frontend/src/pages/projects/dataConnections/const.ts (hunk 5 above)
Comment (manaswinidas) [reply frontend/src/pages/projects/dataConnections/const.ts]: Sorry, I totally misunderstood it. Thanks for pointing it out.

File: frontend/src/pages/projects/dataConnections/const.ts (hunk 7 above)
Comment (manaswinidas) [reply frontend/src/pages/projects/dataConnections/const.ts:64]: value is a required field, I've removed the value's value though.

File: frontend/src/pages/projects/dataConnections/const.ts (hunk 6 above)
Comment (manaswinidas) [reply frontend/src/pages/projects/dataConnections/const.ts:68]: value is a required field, I've removed the value's value though.

Comment (Gkrumbach07) [reply frontend/src/pages/projects/dataConnections/const.ts:68]: yep, you are right. i was looking at the wrong object

File: frontend/src/pages/projects/dataConnections/const.ts (hunk 7 above)
Comment (Gkrumbach07) [reply frontend/src/pages/projects/dataConnections/const.ts:64]: yep, you are right. keep it as is

**Comment by Gkrumbach07** [comment]
Looks good. only concern is that [this](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/pages/projects/screens/detail/data-connections/utils.ts#L62-L69) still will always default to `AWS S3`

```ts
export const getDataConnectionProvider = (dataConnection: DataConnection): string => {
  switch (dataConnection.type) {
    case DataConnectionType.AWS:
      return 'AWS S3';
    default:
      throw new Error('Invalid data connection type');
  }
};
```

@andrewballantyne is this ok

**Comment by Gkrumbach07** [comment]
/lgtm

The provider logic doesn't need to be changed

**Comment by andrewballantyne** [comment]
/unhold

Nevermind 😄

**Comment by shalberd** [comment]
@manaswinidas thank you. Yeah, the backend code libraries have AWS* in their env variables, even though they can work with all S3 compatible providers, that is ok, though.

File: frontend/src/__tests__/integration/pages/modelServing/ModelServingGlobal.spec.ts [hunk 8]
```diff
@@ -41,6 +41,7 @@ test('Edit model', async ({ page }) => {
   await page
     .getByRole('textbox', { name: 'Field list AWS_SECRET_ACCESS_KEY' })
```
Comment (bdattoma) [inline frontend/src/__tests__/integration/pages/modelServing/ModelServingGlobal.spec.ts:42]: in order to make the S3 model source AWS-unspecific, shouldn't you change the field name besides the displayed name?  @manaswinidas

**Pull request merged**